src/other/Table.js

```diff
@@ -424,45 +424,72 @@
                 var rowSelection = context.table.selectAll("tbody > tr > td:first-child");
                 rowWrapperWidth = context.fixedColumn() ? rowSelection.node().offsetWidth : 0;
                 var theadWidth = parseInt(rowWrapperWidth) + parseInt(2 * borderWidth);
-                var rowContents = "<thead>";
-                var tempTrWrapper = document.createElement("div");
-                
+
+                var tHead = rowLabelsWrapper.selectAll("thead").data([0]);
                 if (context.showHeader()) {
-                    var theadTr = document.createElement("tr");
-                    theadTr.style.backgroundColor = context.theadRowBackgroundColor();
-                    var th = document.createElement("th");
-                    th.style.width = theadWidth + "px";
-                    th.style.color = context.theadFontColor();
-                    th.style.fontSize = context.theadFontSize();
-                    th.style.fontFamily = context.theadFontFamily();
-                    th.style.borderColor = context.theadCellBorderColor();
-                    th.innerHTML = origThead.filter(function (d, idx) { return idx === 0; }).html();
-                    theadTr.appendChild(th);
-                    tempTrWrapper.appendChild(theadTr);
-                    rowContents += tempTrWrapper.innerHTML;
-                    
-                } else {
-                    tableMarginHeight = 0;
-                }
-                rowContents += "</thead>";
-                rowSelection.each(function () {
-                    var td = document.createElement("td");
-                    td.className = "row-label";
-                    td.style.color = context.tbodyFontColor();
-                    td.style.fontSize = context.tbodyFontSize();
-                    td.style.fontFamily = context.tbodyFontFamily();
-                    td.style.borderColor = context.tbodyCellBorderColor();
-                    td.innerHTML = this.innerHTML ? this.innerHTML : "&nbsp;";
-                    var tempDiv = document.createElement("div");
-                    tempDiv.appendChild(td);
-                    rowContents += "<tr>" + tempDiv.innerHTML + "</tr>";
-                });
-                rowLabelsWrapper.html(rowContents)
+                    tHead.enter()
+                        .insert("thead", "tbody")
+                        .append("tr")
+                        .style("background-color", context.theadRowBackgroundColor())
+                    ;
```

Comment:
You could have made this:
```
var tHead_tr = tHead.enter()
                        .insert("thead", "tbody")
                        .append("tr")
                        .style("background-color", context.theadRowBackgroundColor())
                    ;
```
Which would then simplify the selectAll below (just a FYI).